src/main/java/net/sf/jabref/gui/keyboard/KeyBindingsDialogViewModel.java

```diff
@@ -0,0 +1,210 @@
+/*  Copyright (C) 2003-2015 JabRef contributors.
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+package net.sf.jabref.gui.keyboard;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.gui.FXAlert;
+import net.sf.jabref.logic.l10n.Localization;
+
+import javafx.beans.property.SimpleStringProperty;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.Alert.AlertType;
+import javafx.scene.control.Button;
+import javafx.scene.control.ButtonBar;
+import javafx.scene.control.ButtonBar.ButtonData;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.SelectionMode;
+import javafx.scene.control.TreeItem;
+import javafx.scene.control.TreeTableColumn;
+import javafx.scene.control.TreeTableView;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyEvent;
+import javafx.stage.Stage;
+
+public class KeyBindingsDialogViewModel {
+
+    private KeyBindingRepository keyBindingRepository;
+
+    @FXML
+    private TreeTableView<KeyBindingWrapper> keyBindingsTable;
+
+    @FXML
+    private TreeTableColumn<KeyBindingWrapper, String> actionColumn;
+
+    @FXML
+    private TreeTableColumn<KeyBindingWrapper, String> shortcutColumn;
+
+    @FXML
+    private Button closeButton;
+
+    @FXML
+    private Button resetButton;
+
+
+    @FXML
+    private void initialize() {
+        keyBindingsTable.getSelectionModel().setSelectionMode(SelectionMode.SINGLE);
+        ButtonBar.setButtonData(resetButton, ButtonData.LEFT);
+    }
+
+    public void initializeView() {
```

Comment:
This is because the keyBindingRepository is not yet initialized, right?
In this case, just call the initializeView in setKeyBindingPreferences (and make it private)